Safety output Unmarshall issues

After running huskyCI in a Python project, the following error is being shown:

```
[HUSKYCI] [!]: Warning: unpinned requirement 'Flask' found in safety_analysis_requirements.txt, unable to check.
[HUSKYCI] [!]: Warning: unpinned requirement 'gunicorn' found in safety_analysis_requirements.txt, unable to check.
[HUSKYCI] [!]: Warning: unpinned requirement 'requests' found in safety_analysis_requirements.txt, unable to check.
[HUSKYCI][ERROR] Could not Unmarshal safetyOutput! invalid character 'W' looking for beginning of value
```

--- a/Makefile
+++ b/Makefile
@@ -64,6 +64,10 @@ build:
 build-client:
 	cd client/cmd && $(GO) build -o "$(HUSKYCICLIENTBIN)" && mv "$(HUSKYCICLIENTBIN)" ../..
 
+## Builds client to the executable file huskyci-client
+build-client-linux:
+	cd client/cmd && GOOS=linux GOARCH=amd64 $(GO) build -o "$(HUSKYCICLIENTBIN)" && mv "$(HUSKYCICLIENTBIN)" ../..
+
 ## Runs huskyci-client
 run-client: build-client
 	./"$(HUSKYCICLIENTBIN)"

--- a/api/analysis/safety.go
+++ b/api/analysis/safety.go
@@ -5,12 +5,12 @@
 package analysis
 
 import (
-	"bufio"
 	"encoding/json"
 	"fmt"
 	"strings"
 
 	"github.com/globocom/huskyCI/api/log"
+	"github.com/globocom/huskyCI/util"
 	"gopkg.in/mgo.v2/bson"
 )
 
@@ -31,18 +31,27 @@ type SafetyIssue struct {
 //SafetyStartAnalysis analyses the output from Safety and sets cResult based on it.
 func SafetyStartAnalysis(CID string, cOutput string) {
 
-	var cResult string
+	var outputJSON string
 	analysisQuery := map[string]interface{}{"containers.CID": CID}
 
-	warningFound := strings.Contains(cOutput, "Warning: unpinned requirement ")
-	if warningFound {
-		tmpcOutput := StringToLastLine(cOutput)
-		cOutput = tmpcOutput
+	requirementsNotFound := strings.Contains(cOutput, "ERROR_REQ_NOT_FOUND")
+	if requirementsNotFound {
+		updateContainerAnalysisQuery := bson.M{
+			"$set": bson.M{
+				"containers.$.cOutput": "Requirements not found or this project uses latest dependencies.",
+				"containers.$.cResult": "warning", // will not fail CI now
+			},
+		}
+		err := UpdateOneDBAnalysisContainer(analysisQuery, updateContainerAnalysisQuery)
+		if err != nil {
+			log.Warning("SafetyStartAnalysis", "SAFETY", 2007, err)
+		}
+		return
 	}
 
-	// step 0.1: error cloning repository!
-	if strings.Contains(cOutput, "ERROR_CLONING") {
-		errorOutput := fmt.Sprintf("Container error: %s", cOutput)
+	errorCloning := strings.Contains(cOutput, "ERROR_CLONING")
+	if errorCloning {
+		errorOutput := fmt.Sprintf("Error cloning: %s", cOutput)
 		updateContainerAnalysisQuery := bson.M{
 			"$set": bson.M{
 				"containers.$.cOutput": errorOutput,
@@ -56,19 +65,40 @@ func SafetyStartAnalysis(CID string, cOutput string) {
 		return
 	}
 
-	// step 1: Unmarshall cOutput into safetyOutput struct.
+	warningFound := strings.Contains(cOutput, "Warning: unpinned requirement ")
+	if warningFound {
+		outputJSON = util.GetLastLine(cOutput)
+		cOutput = outputJSON
+	}
+
+	cOutputSanitized := util.SanitizeSafetyJSON(cOutput)
+
 	safetyOutput := SafetyOutput{}
-	err := json.Unmarshal([]byte(cOutput), &safetyOutput)
+	err := json.Unmarshal([]byte(cOutputSanitized), &safetyOutput)
 	if err != nil {
 		log.Error("SafetyStartAnalysis", "SAFETY", 1018, cOutput, err)
 		return
 	}
 
-	// step 1.1: Sets the container output to "No issues found" if SafetyIssues returns an empty slice
-	if len(safetyOutput.SafetyIssues) == 0 && !warningFound {
+	if len(safetyOutput.SafetyIssues) == 0 {
+
+		if warningFound {
+			updateContainerAnalysisQuery := bson.M{
+				"$set": bson.M{
+					"containers.$.cResult": "warning",
+				},
+			}
+			err := UpdateOneDBAnalysisContainer(analysisQuery, updateContainerAnalysisQuery)
+			if err != nil {
+				log.Error("SafetyStartAnalysis", "SAFETY", 2007, err)
+			}
+			return
+		}
+
 		updateContainerAnalysisQuery := bson.M{
 			"$set": bson.M{
 				"containers.$.cOutput": "No issues found.",
+				"containers.$.cResult": "passed",
 			},
 		}
 		err := UpdateOneDBAnalysisContainer(analysisQuery, updateContainerAnalysisQuery)
@@ -78,42 +108,16 @@ func SafetyStartAnalysis(CID string, cOutput string) {
 		return
 	}
 
-	// step 2: finds Vulnerabilities
-	cResult = "passed"
-	if len(safetyOutput.SafetyIssues) != 0 {
-		cResult = "failed"
-	}
-
-	// step 3: update analysis' cResult into AnalyisCollection.
+	// issues found. client will have to handle with warnings and issues.
 	updateContainerAnalysisQuery := bson.M{
 		"$set": bson.M{
-			"containers.$.cResult": cResult,
+			"containers.$.cResult": "failed",
 		},
 	}
 	err = UpdateOneDBAnalysisContainer(analysisQuery, updateContainerAnalysisQuery)
 	if err != nil {
 		log.Error("SafetyStartAnalysis", "SAFETY", 2007, err)
-		return
 	}
-}
+	return
 
-// StringToLastLine receives a string with multiple lines and returns it's last
-func StringToLastLine(s string) string {
-	var lines []string
-	scanner := bufio.NewScanner(strings.NewReader(s))
-	for scanner.Scan() {
-		lines = append(lines, scanner.Text())
-	}
-	return lines[len(lines)-1]
-}
-
-// GetAllLinesButLast receives a string with multiple lines and returns all but the last line.
-func GetAllLinesButLast(s string) []string {
-	var lines []string
-	scanner := bufio.NewScanner(strings.NewReader(s))
-	for scanner.Scan() {
-		lines = append(lines, scanner.Text())
-	}
-	lines = lines[:len(lines)-1]
-	return lines
 }

--- a/api/config.yaml
+++ b/api/config.yaml
@@ -121,9 +121,9 @@ safety:
     git clone -b %GIT_BRANCH% --single-branch %GIT_REPO% code --quiet 2> /tmp/errorGitCloneSafety
     if [ $? -eq 0 ]; then
       cd code
-      cat *requirements* | grep '=' | grep -v '#' > safety_analysis_requirements.txt 2> /tmp/errorGetRequirements
+      cat *requirements* | grep '=' | grep -v '#' > safety_huskyci_analysis_requirements.txt 2> /tmp/errorGetRequirements
       if [ $? -eq 0 ]; then
-        safety check -r safety_analysis_requirements.txt --json > safety_analysis_output.json 2> /tmp/errorUnpinnedReq
+        safety check -r safety_analysis_requirements.txt --json > safety_huskyci_analysis_output.json 2> /tmp/errorUnpinnedReq
         chmod +x /script.sh
         /script.sh /code/safety_analysis_output.json
         cat /tmp/errorUnpinnedReq 2> /dev/null

--- a/api/log/messagecodes.go
+++ b/api/log/messagecodes.go
@@ -40,7 +40,7 @@ var MsgCode = map[int]string{
 	1015: "Received an invalid repository JSON: ",
 	1016: "Received an invalid repository URL: ",
 	1017: "Received an invalid repository branch: ",
-	1018: "Could not Unmarshall the following safetyOutput; ",
+	1018: "Could not Unmarshall the following safetyOutput: ",
 
 	// MongoDB infos
 	21: "Connecting to MongoDB.",

--- a/client/analysis/analysis.go
+++ b/client/analysis/analysis.go
@@ -106,6 +106,6 @@ func MonitorAnalysis(RID string) (types.Analysis, error) {
 func AnalyzeResult(analysisResult types.Analysis) {
 	fmt.Println()
 	for _, container := range analysisResult.Containers {
-		CheckContainerOutput(container)
+		CheckMongoDBContainerOutput(container)
 	}
 }

--- a/client/analysis/output.go
+++ b/client/analysis/output.go
@@ -11,12 +11,12 @@ import (
 	"strings"
 
 	"github.com/fatih/color"
-	"github.com/globocom/huskyCI/api/analysis"
 	"github.com/globocom/huskyCI/client/types"
+	"github.com/globocom/huskyCI/util"
 )
 
-// CheckContainerOutput will validate the output of a given container.
-func CheckContainerOutput(container types.Container) {
+// CheckMongoDBContainerOutput will validate the output of a given container.
+func CheckMongoDBContainerOutput(container types.Container) {
 
 	switch container.SecurityTest.Name {
 	case "enry":
@@ -37,9 +37,9 @@ func CheckContainerOutput(container types.Container) {
 }
 
 // PrintGosecOutput will print the Gosec output.
-func PrintGosecOutput(containerOutput string) {
+func PrintGosecOutput(mongoDBcontainerOutput string) {
 
-	if containerOutput == "No issues found." {
+	if mongoDBcontainerOutput == "No issues found." {
 		color.Green("[HUSKYCI][*] Gosec :)\n\n")
 		return
 	}
@@ -47,9 +47,9 @@ func PrintGosecOutput(containerOutput string) {
 	foundVuln := false
 	foundInfo := false
 	gosecOutput := types.GosecOutput{}
-	err := json.Unmarshal([]byte(containerOutput), &gosecOutput)
+	err := json.Unmarshal([]byte(mongoDBcontainerOutput), &gosecOutput)
 	if err != nil {
-		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal gosecOutput!", containerOutput)
+		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal gosecOutput!", mongoDBcontainerOutput)
 		os.Exit(1)
 	}
 
@@ -104,9 +104,9 @@ func PrintGosecOutput(containerOutput string) {
 }
 
 // PrintBanditOutput will print Bandit output.
-func PrintBanditOutput(containerOutput string) {
+func PrintBanditOutput(mongoDBcontainerOutput string) {
 
-	if containerOutput == "No issues found." {
+	if mongoDBcontainerOutput == "No issues found." {
 		color.Green("[HUSKYCI][*] Bandit :)\n\n")
 		return
 	}
@@ -114,9 +114,9 @@ func PrintBanditOutput(containerOutput string) {
 	foundVuln := false
 	foundInfo := false
 	banditOutput := types.BanditOutput{}
-	err := json.Unmarshal([]byte(containerOutput), &banditOutput)
+	err := json.Unmarshal([]byte(mongoDBcontainerOutput), &banditOutput)
 	if err != nil {
-		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal banditOutput!", containerOutput)
+		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal banditOutput!", mongoDBcontainerOutput)
 		os.Exit(1)
 	}
 
@@ -171,9 +171,9 @@ func PrintBanditOutput(containerOutput string) {
 }
 
 // PrintRetirejsOutput will print Retirejs output.
-func PrintRetirejsOutput(containerOutput string) {
+func PrintRetirejsOutput(mongoDBcontainerOutput string) {
 
-	if containerOutput == "No issues found." {
+	if mongoDBcontainerOutput == "No issues found." {
 		color.Green("[HUSKYCI][*] RetireJS :)\n\n")
 		return
 	}
@@ -181,9 +181,9 @@ func PrintRetirejsOutput(containerOutput string) {
 	foundVuln := false
 	foundInfo := false
 	retirejsOutput := types.RetirejsOutput{}
-	err := json.Unmarshal([]byte(containerOutput), &retirejsOutput)
+	err := json.Unmarshal([]byte(mongoDBcontainerOutput), &retirejsOutput)
 	if err != nil {
-		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal retirejsOutput!", containerOutput)
+		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal retirejsOutput!", mongoDBcontainerOutput)
 		os.Exit(1)
 	}
 
@@ -250,8 +250,8 @@ func PrintRetirejsOutput(containerOutput string) {
 }
 
 // PrintBrakemanOutput will print Brakeman output.
-func PrintBrakemanOutput(containerOutput string) {
-	if containerOutput == "No issues found." {
+func PrintBrakemanOutput(mongoDBcontainerOutput string) {
+	if mongoDBcontainerOutput == "No issues found." {
 		color.Green("[HUSKYCI][*] Brakeman :)\n\n")
 		return
 	}
@@ -259,9 +259,9 @@ func PrintBrakemanOutput(containerOutput string) {
 	foundVuln := false
 	foundInfo := false
 	brakemanOutput := types.BrakemanOutput{}
-	err := json.Unmarshal([]byte(containerOutput), &brakemanOutput)
+	err := json.Unmarshal([]byte(mongoDBcontainerOutput), &brakemanOutput)
 	if err != nil {
-		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal brakemanOutput!", containerOutput)
+		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal brakemanOutput!", mongoDBcontainerOutput)
 		os.Exit(1)
 	}
 
@@ -315,44 +315,55 @@ func PrintBrakemanOutput(containerOutput string) {
 }
 
 // PrintSafetyOutput will print Safety output.
-func PrintSafetyOutput(containerOutput string) {
+func PrintSafetyOutput(mongoDBcontainerOutput string) {
 
-	if strings.Contains(containerOutput, "ERROR_REQ_NOT_FOUND") {
-		color.Red("[HUSKYCI][X] huskyCI couldn't find any requirements file...\n")
-		color.Red("[HUSKYCI][*] Safety :(\n\n")
-		return
-	}
-
-	// Safety might return a JSON with the "\" and "\"" characters, which needs to be sanitized to be unmarshalled correctly
-	sanitizateContainerOutput := strings.Replace(containerOutput, "\\", "\\\\", -1)
-	sanitizateContainerOutput2 := strings.Replace(sanitizateContainerOutput, "\\\"", "\\\\\"", -1)
-
-	// Safety container returns warnings and the json output in the same string, which need to be split
-	warningFound := strings.Contains(containerOutput, "Warning: unpinned requirement ")
-	if warningFound {
-		tmpcOutput := analysis.StringToLastLine(containerOutput)
-		warningOutput := analysis.GetAllLinesButLast(containerOutput)
-		containerOutput = tmpcOutput
-		for _, warning := range warningOutput {
-			color.Yellow("[HUSKYCI][!]: %s", warning)
-		}
-	}
-
-	if containerOutput == "No issues found." && !warningFound {
+	if mongoDBcontainerOutput == "No issues found." {
 		color.Green("[HUSKYCI][*] Safety :)\n\n")
 		return
 	}
 
-	safetyOutput := types.SafetyOutput{}
-	err := json.Unmarshal([]byte(sanitizateContainerOutput2), &safetyOutput)
-	if err != nil {
-		fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal safetyOutput!", err)
-		os.Exit(1)
+	if mongoDBcontainerOutput == "Requirements not found or this project uses latest dependencies." {
+		fmt.Printf("[HUSKYCI][*] Requirements not found or this project uses latest dependencies.\n")
+		fmt.Printf("[HUSKYCI][*] Safety :|\n\n")
+		return
 	}
 
-	foundVuln := false
+	// Safety returns warnings and the json output in the same string, which need to be split
+	var cOutputSanitized string
+	safetyOutput := types.SafetyOutput{}
+	warningFound := strings.Contains(mongoDBcontainerOutput, "Warning: unpinned requirement ")
+	if !warningFound {
+		// only issue found
+		cOutputSanitized = util.SanitizeSafetyJSON(mongoDBcontainerOutput)
+		err := json.Unmarshal([]byte(cOutputSanitized), &safetyOutput)
+		if err != nil {
+			fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal safetyOutput: ", err)
+			os.Exit(1)
+		}
+	} else {
+		// issues and warnings found
+		onlyWarning := false
+		outputJSON := util.GetLastLine(mongoDBcontainerOutput)
+		outputWarnings := util.GetAllLinesButLast(mongoDBcontainerOutput)
+		cOutputSanitized = util.SanitizeSafetyJSON(outputJSON)
+		err := json.Unmarshal([]byte(cOutputSanitized), &safetyOutput)
+		if err != nil {
+			fmt.Println("[HUSKYCI][ERROR] Could not Unmarshal safetyOutput: ", err)
+			os.Exit(1)
+		}
+		if len(safetyOutput.SafetyIssues) == 0 {
+			onlyWarning = true
+		}
+		for _, warning := range outputWarnings {
+			color.Yellow("[HUSKYCI][!]: %s", warning)
+		}
+		if onlyWarning {
+			fmt.Printf("[HUSKYCI][*] Safety :|\n\n")
+			return
+		}
+	}
+
 	for _, issue := range safetyOutput.SafetyIssues {
-		foundVuln = true
 		color.Red("[HUSKYCI][!] Vulnerable Dependency: %s", issue.Dependency)
 		color.Red("[HUSKYCI][!] Vulnerable Below: %s", issue.Below)
 		color.Red("[HUSKYCI][!] Current Version: %s", issue.Version)
@@ -360,12 +371,6 @@ func PrintSafetyOutput(containerOutput string) {
 		fmt.Println()
 	}
 
-	if foundVuln {
-		color.Red("[HUSKYCI][X] Safety :(\n\n")
-		types.FoundVuln = true
-		return
-	}
-
-	fmt.Printf("[HUSKYCI][*] Safety :|\n\n")
-
+	color.Red("[HUSKYCI][X] Safety :(\n\n")
+	types.FoundVuln = true
 }

--- a/client/analysis/output.go
+++ b/client/analysis/output.go
@@ -192,12 +192,12 @@ func PrintRetirejsOutput(containerOutput string) {
 			for _, vulnerability := range result.RetirejsVulnerabilities {
 				if vulnerability.Severity == "high" {
 					foundVuln = true
-					color.Red("[HUSKYCI] [!] Severity: %s", vulnerability.Severity)
-					color.Red("[HUSKYCI] [!] Details: %s", vulnerability.Info)
-					color.Red("[HUSKYCI] [!] File: %s", issue.File)
-					color.Red("[HUSKYCI] [!] Component: %s", result.Component)
-					color.Red("[HUSKYCI] [!] Version: %s", result.Version)
-					color.Red("[HUSKYCI] [!] Vulnerable Below: %s", vulnerability.Below)
+					color.Red("[HUSKYCI][!] Severity: %s", vulnerability.Severity)
+					color.Red("[HUSKYCI][!] Details: %s", vulnerability.Info)
+					color.Red("[HUSKYCI][!] File: %s", issue.File)
+					color.Red("[HUSKYCI][!] Component: %s", result.Component)
+					color.Red("[HUSKYCI][!] Version: %s", result.Version)
+					color.Red("[HUSKYCI][!] Vulnerable Below: %s", vulnerability.Below)
 					fmt.Println()
 				}
 			}
@@ -209,12 +209,12 @@ func PrintRetirejsOutput(containerOutput string) {
 			for _, vulnerability := range result.RetirejsVulnerabilities {
 				if vulnerability.Severity == "medium" {
 					foundVuln = true
-					color.Yellow("[HUSKYCI] [!] Severity: %s", vulnerability.Severity)
-					color.Yellow("[HUSKYCI] [!] Details: %s", vulnerability.Info)
-					color.Yellow("[HUSKYCI] [!] File: %s", issue.File)
-					color.Yellow("[HUSKYCI] [!] Component: %s", result.Component)
-					color.Yellow("[HUSKYCI] [!] Version: %s", result.Version)
-					color.Yellow("[HUSKYCI] [!] Vulnerable Below: %s", vulnerability.Below)
+					color.Yellow("[HUSKYCI][!] Severity: %s", vulnerability.Severity)
+					color.Yellow("[HUSKYCI][!] Details: %s", vulnerability.Info)
+					color.Yellow("[HUSKYCI][!] File: %s", issue.File)
+					color.Yellow("[HUSKYCI][!] Component: %s", result.Component)
+					color.Yellow("[HUSKYCI][!] Version: %s", result.Version)
+					color.Yellow("[HUSKYCI][!] Vulnerable Below: %s", vulnerability.Below)
 					fmt.Println()
 				}
 			}
@@ -226,12 +226,12 @@ func PrintRetirejsOutput(containerOutput string) {
 			for _, vulnerability := range result.RetirejsVulnerabilities {
 				if vulnerability.Severity == "low" {
 					foundInfo = true
-					color.Blue("[HUSKYCI] [!] Severity: %s", vulnerability.Severity)
-					color.Blue("[HUSKYCI] [!] Details: %s", vulnerability.Info)
-					color.Blue("[HUSKYCI] [!] File: %s", issue.File)
-					color.Blue("[HUSKYCI] [!] Component: %s", result.Component)
-					color.Blue("[HUSKYCI] [!] Version: %s", result.Version)
-					color.Blue("[HUSKYCI] [!] Vulnerable Below: %s", vulnerability.Below)
+					color.Blue("[HUSKYCI][!] Severity: %s", vulnerability.Severity)
+					color.Blue("[HUSKYCI][!] Details: %s", vulnerability.Info)
+					color.Blue("[HUSKYCI][!] File: %s", issue.File)
+					color.Blue("[HUSKYCI][!] Component: %s", result.Component)
+					color.Blue("[HUSKYCI][!] Version: %s", result.Version)
+					color.Blue("[HUSKYCI][!] Vulnerable Below: %s", vulnerability.Below)
 					fmt.Println()
 				}
 			}
@@ -268,37 +268,37 @@ func PrintBrakemanOutput(containerOutput string) {
 	for _, warning := range brakemanOutput.Warnings {
 		if warning.Confidence == "High" {
 			foundVuln = true
-			color.Red("[HUSKYCI] [!] Confidence: %s", warning.Confidence)
-			color.Red("[HUSKYCI] [!] Type: %s", warning.Type)
-			color.Red("[HUSKYCI] [!] Details: %s", warning.Details)
-			color.Red("[HUSKYCI] [!] Info: %s", warning.Message)
-			color.Red("[HUSKYCI] [!] File: %s", warning.File)
-			color.Red("[HUSKYCI] [!] line: %d", warning.Line)
-			color.Red("[HUSKYCI] [!] Code: %s", warning.Code)
+			color.Red("[HUSKYCI][!] Confidence: %s", warning.Confidence)
+			color.Red("[HUSKYCI][!] Type: %s", warning.Type)
+			color.Red("[HUSKYCI][!] Details: %s", warning.Details)
+			color.Red("[HUSKYCI][!] Info: %s", warning.Message)
+			color.Red("[HUSKYCI][!] File: %s", warning.File)
+			color.Red("[HUSKYCI][!] line: %d", warning.Line)
+			color.Red("[HUSKYCI][!] Code: %s", warning.Code)
 			fmt.Println()
 		}
 
 		if warning.Confidence == "Medium" {
 			foundVuln = true
-			color.Yellow("[HUSKYCI] [!] Confidence: %s", warning.Confidence)
-			color.Yellow("[HUSKYCI] [!] Type: %s", warning.Type)
-			color.Yellow("[HUSKYCI] [!] Details: %s", warning.Details)
-			color.Yellow("[HUSKYCI] [!] Info: %s", warning.Message)
-			color.Yellow("[HUSKYCI] [!] File: %s", warning.File)
-			color.Yellow("[HUSKYCI] [!] line: %d", warning.Line)
-			color.Yellow("[HUSKYCI] [!] Code: %s", warning.Code)
+			color.Yellow("[HUSKYCI][!] Confidence: %s", warning.Confidence)
+			color.Yellow("[HUSKYCI][!] Type: %s", warning.Type)
+			color.Yellow("[HUSKYCI][!] Details: %s", warning.Details)
+			color.Yellow("[HUSKYCI][!] Info: %s", warning.Message)
+			color.Yellow("[HUSKYCI][!] File: %s", warning.File)
+			color.Yellow("[HUSKYCI][!] line: %d", warning.Line)
+			color.Yellow("[HUSKYCI][!] Code: %s", warning.Code)
 			fmt.Println()
 		}
 
 		if warning.Confidence == "Low" {
 			foundInfo = true
-			color.Blue("[HUSKYCI] [!] Confidence: %s", warning.Confidence)
-			color.Blue("[HUSKYCI] [!] Type: %s", warning.Type)
-			color.Blue("[HUSKYCI] [!] Details: %s", warning.Details)
-			color.Blue("[HUSKYCI] [!] Info: %s", warning.Message)
-			color.Blue("[HUSKYCI] [!] File: %s", warning.File)
-			color.Blue("[HUSKYCI] [!] line: %d", warning.Line)
-			color.Blue("[HUSKYCI] [!] Code: %s", warning.Code)
+			color.Blue("[HUSKYCI][!] Confidence: %s", warning.Confidence)
+			color.Blue("[HUSKYCI][!] Type: %s", warning.Type)
+			color.Blue("[HUSKYCI][!] Details: %s", warning.Details)
+			color.Blue("[HUSKYCI][!] Info: %s", warning.Message)
+			color.Blue("[HUSKYCI][!] File: %s", warning.File)
+			color.Blue("[HUSKYCI][!] line: %d", warning.Line)
+			color.Blue("[HUSKYCI][!] Code: %s", warning.Code)
 			fmt.Println()
 		}
 	}
@@ -334,7 +334,7 @@ func PrintSafetyOutput(containerOutput string) {
 		warningOutput := analysis.GetAllLinesButLast(containerOutput)
 		containerOutput = tmpcOutput
 		for _, warning := range warningOutput {
-			color.Yellow("[HUSKYCI] [!]: %s", warning)
+			color.Yellow("[HUSKYCI][!]: %s", warning)
 		}
 	}
 
@@ -353,10 +353,10 @@ func PrintSafetyOutput(containerOutput string) {
 	foundVuln := false
 	for _, issue := range safetyOutput.SafetyIssues {
 		foundVuln = true
-		color.Red("[HUSKYCI] [!] Vulnerable Dependency: %s", issue.Dependency)
-		color.Red("[HUSKYCI] [!] Vulnerable Below: %s", issue.Below)
-		color.Red("[HUSKYCI] [!] Current Version: %s", issue.Version)
-		color.Red("[HUSKYCI] [!] Comment: %s", issue.Comment)
+		color.Red("[HUSKYCI][!] Vulnerable Dependency: %s", issue.Dependency)
+		color.Red("[HUSKYCI][!] Vulnerable Below: %s", issue.Below)
+		color.Red("[HUSKYCI][!] Current Version: %s", issue.Version)
+		color.Red("[HUSKYCI][!] Comment: %s", issue.Comment)
 		fmt.Println()
 	}
 

--- a/util/util.go
+++ b/util/util.go
@@ -1,6 +1,7 @@
 package util
 
 import (
+	"bufio"
 	"crypto/tls"
 	"crypto/x509"
 	"io/ioutil"
@@ -69,4 +70,33 @@ func HandleCmd(repositoryURL, repositoryBranch, cmd string) string {
 func HandlePrivateSSHKey(rawString string) string {
 	cmdReplaced := strings.Replace(rawString, "GIT_PRIVATE_SSH_KEY", os.Getenv("GIT_PRIVATE_SSH_KEY"), -1)
 	return cmdReplaced
+}
+
+// GetLastLine receives a string with multiple lines and returns it's last
+func GetLastLine(s string) string {
+	var lines []string
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	return lines[len(lines)-1]
+}
+
+// GetAllLinesButLast receives a string with multiple lines and returns all but the last line.
+func GetAllLinesButLast(s string) []string {
+	var lines []string
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	lines = lines[:len(lines)-1]
+	return lines
+}
+
+// SanitizeSafetyJSON returns a sanitized string from Safety container logs.
+// Safety might return a JSON with the "\" and "\"" characters, which needs to be sanitized to be unmarshalled correctly.
+func SanitizeSafetyJSON(s string) string {
+	s1 := strings.Replace(s, "\\", "\\\\", -1)
+	s2 := strings.Replace(s1, "\\\"", "\\\\\"", -1)
+	return s2
 }

